scripts/mkc.sh

```diff
@@ -5,31 +5,23 @@ CERT_PATH=/Users/xcode/certificates.p12
 
 echo "Keychain Starting"
 
-# Code 0 = Ok
-# Code 1 = Locked
-# Code 36 = Locked
-# Code 50 = Not Found
-/usr/bin/security show-keychain-info $KC_NAME 2>/dev/null
+# check if target keychain exists:
+/usr/bin/security list-keychains -d user | grep $KC_NAME &> /dev/null
 rv=$?
 
-if [ $rv -eq 0 ]; then
-  /usr/bin/security list-keychains -d user -s $KC_NAME
-  echo "Keychain ready"
-elif [ $rv -eq 50 ]; then
+if [[ $rv == 0 ]]; then
+  # unlock keychain as it's always locked after reboot:
+  echo "Unlocking keychain"
+  /usr/bin/security unlock-keychain -p $1 $KC_NAME
+  /usr/bin/security show-keychain-info $KC_NAME
+else
   echo "Building keychian"
   /usr/bin/security create-keychain -p $1 $KC_NAME
   /usr/bin/security unlock-keychain -p $1 $KC_NAME
   /usr/bin/security list-keychains -d user -s $KC_NAME
   /usr/bin/security import $CERT_PATH -k $KC_NAME -P $1 -T /usr/bin/codesign
   /usr/bin/security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k $1 $KC_NAME
   /usr/bin/security set-keychain-settings $KC_NAME
-elif [ $rv -eq 1 ] || [ $rv -eq 36 ] ; then
```

Comment:
Why remove the checks for a return value of 50, 1, and 36?

Reply:
@jleach see line-9: instead of `security show-keychain-info` it's updated to `security list-keychains`. I found the issue where even this is running in the background, of the keychain is not yet unlocked, it'll prompt a window for password entry. That explains why we are seeing that from time to time on agent and also why cicd keychain is not unlocked. So there's no return for the Code value anymore, I only check if it exists or not, if yes, always unlock it. If not the create.